crates/debug/src/lib.rs

```diff
@@ -34,10 +30,13 @@ pub fn write_debugsections(obj: &mut Object, sections: Vec<DwarfSection>) -> Res
     for (name, relocs) in relocs {
         let section_id = *ids.get(&name).unwrap();
         for reloc in relocs {
-            let target_symbol = if reloc.target.starts_with("_wasm_function") {
-                obj.symbol_id(reloc.target.as_bytes()).unwrap()
-            } else {
-                obj.section_symbol(*ids.get(&reloc.target).unwrap())
+            let target_symbol = match reloc.target {
+                DwarfSectionRelocTarget::Func(id) => obj
+                    .symbol_id(format!("_wasm_function_{}", id).as_bytes())
```

Comment:
FWIW I've seen this prefix in a number of places. Is there a way that we can canonicalize what's going on here? Ideally the name of a function would be contextually defined on some sort of `Module` type as well to account for the name section, but that may be a bit of a larger refactoring.

Reply:
In this case these names are just unique string identifier (which can be converted back into `FuncIndex`). We can maintain separate map of FuncIndex<->obj function somewhere else.

Reply:
I'm mostly just a bit worried about this conversion is open-coded in a number of places. Not the worst thing in the world but it'd be great if we could centralize/document this somehow.

Reply:
centralized name handling (inside object.rs) and added more docs